lib/index.js

```diff
@@ -153,14 +159,32 @@ internals.initialize = function (server, next) {
 
 internals.stop = function (server, next) {
 
-    const collector = internals.state(server.root).collector;
+    const rootCollector = internals.state(server.root).collector;
 
     // Do not teardown if specifically asked not to
-    if (collector.teardownOnStop === false) {
+    if (rootCollector.teardownOnStop === false) {
         return next();
     }
 
-    return server.root.knex().destroy(next);
+    const knexConnections = [];
+
+    Object.keys(rootCollector.knexGroups).forEach((key) => {
+
+        knexConnections.push(rootCollector.knexGroups[key].knex);
+    });
+
+    // destroy() all knex instances
+    Items.parallel(knexConnections, (item, nxt) => {
+
+        item.destroy(() => {
+
+            nxt();
```

Comment:
It's weird, I was looking on their docs and Knex barely mentions destroy(). It seems it's only useful if you keep 'pools' open, which I don't involve in my tests. See here also no errors seem to be getting passed around if this is the correct destroy definition: https://github.com/tgriesser/knex/blob/a4bfee77513d000cf0fddf130a370071c240f93d/src/client.js#L267-L286